crates/ruma-appservice-api/src/lib.rs

```diff
@@ -16,7 +16,7 @@ pub mod thirdparty;
 /// A namespace defined by an application service.
 ///
 /// Used for [appservice registration](https://spec.matrix.org/v1.2/application-service-api/#registration).
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
```

Comment:
I think we shouldn't have `PartialEq` here at all.

```suggestion
#[derive(Clone, Debug, Serialize, Deserialize)]
```